corehq/apps/geospatial/tasks.py

```diff
@@ -9,3 +28,38 @@ def geo_cases_reassignment_update_owners(domain, case_owner_updates_dict, task_k
     finally:
         celery_task_tracker = CeleryTaskTracker(task_key)
         celery_task_tracker.mark_completed()
+
+
+@serial_task('async-index-es-docs', timeout=30 * 60, queue='background_queue', ignore_result=True)
+def index_es_docs_with_location_props(domain):
+    celery_task_tracker = get_celery_task_tracker(domain, INDEX_ES_TASK_HELPER_BASE_KEY)
+    if celery_task_tracker.is_active():
+        return
+
+    geo_case_prop = get_geo_case_property(domain)
+    query = _es_case_query(domain, geo_case_prop)
+    doc_count = query.count()
+    if doc_count > MAX_GEOSPATIAL_INDEX_DOC_LIMIT:
+        celery_task_tracker.set_message(
+            _('This domain contains too many cases and so they will not be made available '
+              'for use by this feature. Please reach out to support.')
+        )
+        return
+
+    celery_task_tracker.mark_requested()
+    batch_count = get_batch_count(doc_count, DEFAULT_QUERY_LIMIT)
+    try:
+        for i in range(batch_count):
+            progress = (i / batch_count) * 100
+            celery_task_tracker.set_message(
+                _(f'Cases are being made ready for use by this feature. Please be patient. ({progress}%)')
```

Comment:
I wonder if it would be a good idea to just store the progress number and state of the task instead of entire translated message. 
Since HQ translates strings in realtime, so storing the translated messages might not be a good idea?
Or maybe not store a translated message and translate message before rendering it to user?

Reply:
Thinking through this, I'm inclined to agree. We could simply offload the translated messages to the HTML template itself, and then just show the correct message depending on the status and progress of task. I'll update the implementation for this to use progress instead of messages.

Reply:
I have updated the implementation to make use of progress instead of messages ([6906d82](https://github.com/dimagi/commcare-hq/pull/35126/commits/6906d828c78293671103f8edadef77d568694369)).